pkg/storage/mongo/mongo.go

```diff
@@ -2,35 +2,38 @@ package mongo
 
 import (
 	"github.com/globalsign/mgo"
+	"github.com/gomods/athens/pkg/storage/mongoutil"
 )
 
 // ModuleStore represents a mongo backed storage backend.
 type ModuleStore struct {
-	s   *mgo.Session
-	d   string // database
-	c   string // collection
-	url string
+	s           *mgo.Session
+	d           string // database
+	c           string // collection
+	connDetails *mongoutil.ConnDetails
 }
 
 // NewStorage returns an unconnected Mongo backed storage
 // that satisfies the Backend interface.  You must call
 // Connect() on the returned store before using it.
-func NewStorage(url string) *ModuleStore {
-	return &ModuleStore{url: url}
+//
+// TODO: take the database and collection names as parameters
+func NewStorage(connDetails *mongoutil.ConnDetails) *ModuleStore {
+	return &ModuleStore{
+		connDetails: connDetails,
+		d:           "athens",
+		c:           "modules",
+	}
 }
 
-// Connect conntect the the newly created mongo backend.
+// Connect connects the the newly created mongo backend.
 func (m *ModuleStore) Connect() error {
-	s, err := mgo.Dial(m.url)
+	s, err := mongoutil.GetSession(m.connDetails, "athens")
```

Comment:
is "athens" `m.d`?

Reply:
yup. now that you pointed that out, I should pull that out into a constant